bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,512 @@
+                    rhyme_track, unit, is_last_line
+                )
+                lines.append(new_line)
+            else:
+                new_line = await self._get_rhyming_line(
+                    word_rhymes=rhyme_track[unit],
+                    existing_lines=lines,
+                    error_info=(ctx, is_first_error, scheme, unit_count,
+                                time_start)
+                )
+                lines.append(new_line)
+
+                # If last line, it will not be referred to again
+                if not is_last_line:
+                    rhyme_track[unit] |= await self._get_rhyme_set(
+                        instance=self,
+                        word=self._get_last_word(new_line)
+                    )
+
+            unit_count[unit] -= 1
+
+        return lines
+
+    async def send_markov_poem(
+        self,
+        ctx: commands.Context,
+        scheme: str,
+        unit_count: Dict[str, int],
+        time_start: datetime.time,
+        is_first_error: bool = True
+    ) -> None:
+        """Gives the user a paginated embed of their markov poem."""
+        lines = await self._get_generated_markov_poem(
+            instance=self,
+            timeout=self.POEM_TIMEOUT,
+            ctx=ctx,
+            scheme=scheme,
+            unit_count=unit_count,
+            time_start=time_start,
+            is_first_error=is_first_error
+        )
+
+        elapsed_time = datetime.now() - time_start
+        elapsed_time = elapsed_time.seconds
+
+        poem_embed = Embed(
+            title="A Markov Poem For " + str(ctx.author.name),
+            color=Colours.soft_red
+        )
+
+        await LinePaginator.paginate(
+            lines,
+            ctx,
+            poem_embed,
+            max_lines=31,
+            max_size=1000,
+            empty=False,
+            footer_text=f"Elapsed time: {elapsed_time}s\n"
+                        "Rhymes API provided by datamuse."
+        )
+
+    @commands.max_concurrency(1, commands.BucketType.user)
+    @commands.command(
+        aliases=("poem", "mpoem", "m_poem", "markpoem", "mark_poem"),
+        help=f"""This command generates a poem via a markov chain, and all you
+            have to do is provide a rhyme scheme.
+
+            The rhyme scheme is made from characters separated by slashes. E.g
+            "abab/cdcd/efef/gg". The slashes denote a new stanza, i.e, they
+            create an empty line. Same characters mean that the lines rhyme.
+            Note that the characters are case sensitive! For example, a and A
+            represent two different rhyme schemes.
+
+            You may also use our existing rhyme scheme templates:
+            {chr(10).join((k + " --- " + v) for k, v in templates.items())}""",
+        brief="Gives the user a love poem made with a markov chain."
+    )
+    async def markov_poem(
+        self,
+        ctx: commands.Context,
+        rhyme_scheme: str
+    ) -> None:
```

Comment:
We should sanitize the `rhyme_scheme` with regex, as we don't want random characters going into the GET request that goes off here:
https://github.com/python-discord/sir-lancebot/blob/ef61267835e0e37bd4e772fa394e678d857d72c7/bot/exts/valentines/markov_poem_generator.py#L227-L230

Reply:
Hi @Xithrius, thanks for your feedback. What do you mean by sanitize? Is it just to make it so that only letters go through?

Reply:
If there are any characters that aren't letters it would probably return an empty array. We can probably use [`urllib.parse.quote`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote) with a `safe=''` kwarg.

Reply:
I don't really understand this. The characters inside `rhyme_scheme` don't actually end up in `word`, so they wouldn't be going into the GET request. 

Also, using `urllib.parse.quote` would turn one character into many. For instance, let's say someone types in `.poem iñiñ` (which is the same as `.poem abab` except with special characters), the sanitization would turn it into `.poem i%C3%B1i%C3%B1`. So the `ñ` is now a group of 6 letters, and makes a whole new poem entirely.

I apologize if I misunderstood.